glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -129,8 +137,14 @@ open class GleanInternalAPI internal constructor () {
     fun initialize(
         applicationContext: Context,
         uploadEnabled: Boolean,
-        configuration: Configuration = Configuration()
+        configuration: Configuration = Configuration(),
+        buildInfo: BuildInfo? = null
     ) {
+        this.buildInfo = buildInfo
+        if (buildInfo == null) {
+            Log.w(LOG_TAG, "Passing a non-null BuildInfo object will be required in the future. See bug 1691953")
+        }
+
```

Comment:
If we wanted to get fancy, we could probably create an overload for `Glean.initialize` without `buildInfo` (i.e. the current signature) and mark it as [deprecated](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-deprecated/) (which talks nicely to the compiler and Android Studio!).

It's a super nit, not very important, though.

Reply:
This is a good idea.  Might as well...